src/wp-includes/class-wp-textdomain-registry.php

```diff
@@ -32,6 +32,18 @@ class WP_Textdomain_Registry {
 	 */
 	protected $current = array();
 
+	/**
+	 * List of domains and their default language directory path.
+	 *
+	 * @see load_plugin_textdomain()
+	 * @see load_theme_textdomain()
+	 *
+	 * @since 6.1.0
+	 *
+	 * @var array
+	 */
+	protected $default_path = array();
```

Comment:
```suggestion
	protected $default_paths = array();
```
Minor nitpick, would it make sense to use a plural noun here, for consistency with `$cached_mo_files`?

Reply:
> would it make sense to use a plural noun here

It would make more sense to me, yes. 👍

Reply:
Reviewing the entire class further...

* why name it `$default_path` when the class does not also keep "non-default" path(s) elsewhere?
* why use hardcoded paths in `get_path_from_lang_dir()` and not have that info registered in each `$default_path`?

I'm not asking for an explanation 😅 as much as thinking out loud about what was not immediately obvious to me when looking at it for the first time with 0 knowledge about it beforehand 💙

This boils down to an in-memory cache, grouped by: domain, locale, location. Can we just use the existing Cache API for this instead? 😅